### DISCUSSION TITLE
Check my tag type

### DISCUSSION THREAD

**Discussion by sesitcsl2** [ikarus23/MifareClassicTool#424] [General]
Hey, I order a Mifare S50 13.56MHz RFID CUID tags, which should allow rewriting the UID. But so far I am unsucessfull with it. I can write to other sectors but cloning UID feature does not work. Is there way how to check if my Tags are really 2nd gen with changeable UID?
Thanks

**Comment by ikarus23** [comment]
Sorry, I don't know of a way to check what version of magic tag you have with an Android phone. But if the "clone UID" tool does not work, it is most likely not a gen2 card.

The best magic card detection can be found in the Proxmark3 device running https://github.com/RfidResearchGroup/proxmark3.
